src/Microsoft.Extensions.HealthChecks.Abstractions/HealthCheckResult.cs

```diff
@@ -0,0 +1,182 @@
+            => new HealthCheckResult(HealthCheckStatus.Healthy, exception: null, description: description, data: data);
+
+        /// <summary>
+        /// Creates a <see cref="HealthCheckResult"/> representing a component in a degraded state.
+        /// </summary>
+        /// <returns>A <see cref="HealthCheckResult"/> representing a component in a degraded state.</returns>
+        public static HealthCheckResult Degraded()
+            => new HealthCheckResult(HealthCheckStatus.Degraded, exception: null, description: string.Empty, data: null);
+
+        /// <summary>
+        /// Creates a <see cref="HealthCheckResult"/> representing a component in a degraded state.
+        /// </summary>
+        /// <returns>A <see cref="HealthCheckResult"/> representing a component in a degraded state.</returns>
+        /// <param name="description">A human-readable description of the status of the component that was checked.</param>
+        public static HealthCheckResult Degraded(string description)
+            => new HealthCheckResult(HealthCheckStatus.Degraded, exception: null, description: description, data: null);
+
+        /// <summary>
+        /// Creates a <see cref="HealthCheckResult"/> representing a component in a degraded state.
+        /// </summary>
+        /// <returns>A <see cref="HealthCheckResult"/> representing a component in a degraded state.</returns>
+        /// <param name="description">A human-readable description of the status of the component that was checked.</param>
+        /// <param name="data">Additional key-value pairs describing the health of the component</param>
+        public static HealthCheckResult Degraded(string description, IReadOnlyDictionary<string, object> data)
+            => new HealthCheckResult(HealthCheckStatus.Degraded, exception: null, description: description, data: data);
+
+        /// <summary>
+        /// Creates a <see cref="HealthCheckResult"/> representing a component in a degraded state.
+        /// </summary>
+        /// <returns>A <see cref="HealthCheckResult"/> representing a component in a degraded state.</returns>
+        public static HealthCheckResult Degraded(Exception exception)
+            => new HealthCheckResult(HealthCheckStatus.Degraded, exception: null, description: string.Empty, data: null);
+
+        /// <summary>
+        /// Creates a <see cref="HealthCheckResult"/> representing a component in a degraded state.
+        /// </summary>
+        /// <returns>A <see cref="HealthCheckResult"/> representing a component in a degraded state.</returns>
+        /// <param name="description">A human-readable description of the status of the component that was checked.</param>
+        /// <param name="exception">An <see cref="Exception"/> representing the exception that was thrown when checking for status (if any)</param>
+        public static HealthCheckResult Degraded(string description, Exception exception)
```

Comment:
Is this a replacement of warning? I like it

Reply:
Yeah, "Warning" seemed to general.